### ISSUE TITLE
[Bug] Last rule configuration is not considered applied for a file when it is applied to "all files", while previous configuration is applied to only some files.

### ISSUE THREAD

**Issue by michkot** [eslint/config-inspector#151]
Version 1.2.0
eslint 9.34.0

Given the following config

```
import { defineConfig } from 'eslint/config';
import pluginJs from '@eslint/js';

export default defineConfig([
	pluginJs.configs.recommended,
  {name: "with files", files: ["**/*.js"], rules: {"no-debugger": ["warn"]}},
  {name: "without files"                , rules: {"no-debugger": ["off"]}},
]);
```

I am expecting that "after merging, last configuration wins": so specifically that `no-debugger` is off for files, as all three configs (recommended, with, without) apply to *.js files, and for all non *.js files it would be the first and last -> outcome is always off.

However the inspector incorrectly shows that the outcome is warning (notice the warn triangle next to the rule)

<img width="1233" height="795" alt="Image" src="https://github.com/user-attachments/assets/22dc7788-9f45-4b6f-b33c-642d1989a0c5" />

Checking what eslint actually does when i put `debugger;` into the file -> it shows no errors (as expected,).

When I change the last config so it has defined `files`, like this:

> `{name: "without files", files: ["**/*.js"] , rules: {"no-debugger": ["off"]}},`

The inspector correctly shows that the rule will be off for the file:

<img width="816" height="836" alt="Image" src="https://github.com/user-attachments/assets/c8989a83-a1d3-4c94-b5f0-a3d028c310c4" />

I have verified that there is also no issue when both of the rule configurations are applied to all files (no `files: [...]` section is present):

<img width="458" height="376" alt="Image" src="https://github.com/user-attachments/assets/e58c3152-1792-4ac9-9506-3d84dd9f2806" />
